src/containers/App.js

```diff
@@ -70,7 +71,7 @@ function isWebPushSubscribed(endpoint) {
  * See MDN doc about PushSubscription (https://developer.mozilla.org/en-US/docs/Web/API/PushSubscription)
  * @return {Promise}
  */
-function updateSubscriptionOnServer(subscription) {
+function updateSubscriptionOnServer(subscription, userInfo) {
```

Comment:
Can we move those functions with param `userInfo` in to class `App` ?  In this way, we can use `userInfo` by props instead of params passed top-down through many functions.

Reply:
Thanks for your advice,
that would be a quick refactoring.
However, I found that I can use promise chain to address this problem.